video/out/gpu/d3d11_helpers.c

```diff
@@ -78,8 +78,63 @@ static int get_feature_levels(int max_fl, int min_fl,
     return len;
 }
 
-static HRESULT create_device(struct mp_log *log, bool warp, bool debug,
-                             int max_fl, int min_fl, ID3D11Device **dev)
+static IDXGIAdapter1 *get_d3d11_adapter(struct mp_log *log, char *requested_adapter_name)
+{
+    HRESULT hr = S_OK;
+    IDXGIFactory1 *factory;
+    IDXGIAdapter1 *picked_adapter = NULL;
+
+    hr = pCreateDXGIFactory1(&IID_IDXGIFactory1, (void **)&factory);
+    if (FAILED(hr)) {
+        mp_fatal(log, "Failed to create a DXGI factory: %s\n",
+                 mp_HRESULT_to_str(hr));
+        return NULL;
+    }
+
+    for (unsigned int adapter_num = 0; hr != DXGI_ERROR_NOT_FOUND; adapter_num++)
+    {
+        IDXGIAdapter1 *adapter = NULL;
+        DXGI_ADAPTER_DESC1 desc = { 0 };
+        char *adapter_description = NULL;
+
+        hr = IDXGIFactory1_EnumAdapters1(factory, adapter_num, &adapter);
+        if (FAILED(hr)) {
+            if (hr != DXGI_ERROR_NOT_FOUND) {
+                mp_fatal(log, "Failed to enumerate at adapter %u\n",
+                         adapter_num);
+            }
+            continue;
+        }
+
+        if (FAILED(IDXGIAdapter1_GetDesc1(adapter, &desc))) {
+            mp_fatal(log, "Failed to get adapter description when listing at adapter %u\n",
+                     adapter_num);
+            continue;
+        }
+
+        adapter_description = mp_to_utf8(NULL, desc.Description);
+
+        mp_verbose(log, "Adapter %u: vendor: %u, description: %s\n",
+                adapter_num, desc.VendorId,
+                adapter_description ? adapter_description : "<No Description>");
+
+        if ((requested_adapter_name && adapter_description) &&
+            !strcmp(requested_adapter_name, adapter_description)) {
```

Comment:
Minor nit: The first set of (...) is redundant (it's all just '&&'), and it's normally not mpv coding style to leave the { on the same line of there are multiple lines (see contribute.md).